### PR TITLE
Add new module 'zookeeper'

### PR DESCRIPTION
**Pull request by jbowes**
This module allows creating, updating, and removing keys from ZooKeeper
(http://zookeeper.apache.org).

You may optionally specify a list of ACLs to apply to the key, and
either ensure that they exist, or that they are the only ACLs on the
key.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks for submitting this module to Ansible Extras. Apologies that it’s taken a while to get your module reviewed.

To help facilitate reviews, we’ve broadened the number of people who can approve modules for inclusion into the Extras repository. The list of official reviewers can be found here:

https://github.com/ansible/ansible-modules-extras/blob/devel/REVIEWERS.md 

Our new policy is that if a new module is reviewed and approved by at least two official module reviewers, the module will be approved for inclusion.  We will be asking the community of reviewers to take a look at these modules on a regular basis.

To ensure that your module has the best chance of being approved, please double-check that you adhere to the Ansible module guidelines: http://docs.ansible.com/developing_modules.html#module-checklist

File: clustering/zookeeper.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+logging.getLogger("kazoo").propagate = False
+
+try:
+    from kazoo.client import KazooClient
+    from kazoo.exceptions import NoAuthError, NoNodeError, ZookeeperError
+    from kazoo.security import make_acl, make_digest_acl_credential
+except ImportError:
+    print "failed=True msg='kazoo is required for this module. "\
+          "see https://kazoo.readthedocs.org/en/latest/install.html'"
+    sys.exit(1)
```
Comment (bcoca) [inline clustering/zookeeper.py:116]: modules should never exit or throw exceptions, capture the import error and use fail_json to return the message, you can look at ec2/openstack modules for an example

File: clustering/zookeeper.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+# Initialiize Ansible Module
+# import module snippets
+from ansible.module_utils.basic import *
```
Comment (bcoca) [inline clustering/zookeeper.py:120]: you might want to do this at the bottom of the file, this is not a real import and will mess up line numbers for error reporting

**Comment by jbowes** [comment]
Except for ACLs, this is covered by #811. Closing!

**Comment by jcftang** [comment]
-It's not just ACL's that this PR covers, but setting of znodes is covered here that isn't in the znode plugin. It would be nice if the current znode plugin also supported setting of values for znodes-

update: the znode plugin does create/update znodes, it wasn't obvious to me when reading through the docs.

**Pull request closed without merging**